Quiz tests and docs

YuriiHoliuk opened the pull request:
тут є проблема, навіть якщо тести фейляться і виводиться помилка вони всеодно: 4/4 passed
Хтось знає що не так?

YuriiHoliuk committed:
add tests

YuriiHoliuk committed:
add raml doc for endpoint

vnormatov commented on test/quiz.spec.js:
```diff
@@ -0,0 +1,91 @@
+    });
+
+    beforeEach(async () => {
+        await db.removeCollection(collection);
```
может это лучше делать в after

YuriiHoliuk replied:
спеціально зробив в before щоб на тесті мати чисту базу

vnormatov commented on test/quiz.spec.js:
```diff
@@ -0,0 +1,91 @@
+                .get('/tests')
+                .set('Accept', 'application/json')
+                .then(response => {
+                    const quiz = Object.assign({}, response.body[0]);
```
Обычное `const quiz =  response.body[0];` чем не угодило?)

vnormatov commented:
По поводу проблемы при фэйлах тестов, - такая же ситуация, пока хз почему. Предполагаю нужно как то тесты по другому запускать, что б итог считался правильно

YuriiHoliuk committed:
remove object assign

YuriiHoliuk committed:
fix tests

YuriiHoliuk committed:
cleanup

YuriiHoliuk committed:
fix path

YuriiHoliuk committed:
skip quiz

tatsiak committed:
Merge branch 'develop' into quiz-tests-and-docs

YuriiHoliuk committed:
fix part of tests

YuriiHoliuk committed:
add getOne db method & getQuiz method

YuriiHoliuk committed:
get quiz by id in spec

YuriiHoliuk committed:
fix all quiz tests

YuriiHoliuk committed:
revert commit

YuriiHoliuk committed:
---

Commit:
fix db issue on pftServer creation

- add eslint
- refactore code style
- refactore helper
- refactore tests
- refactore pftServer creation

Commit:
Merge branch 'develop' into quiz-tests-and-docs

# Conflicts:
#	lib/routes/index.js
#	lib/routes/quiz.js

Commit:
fix few tests

Commit:
fix getQuizById, fix tests, edit .eslintrc

tatsiak reviewed (approved):
approve
